src/ctelnet.cpp

```diff
@@ -1603,27 +1597,23 @@ int cTelnet::decompressBuffer( char *& in_buffer, int& length, char* out_buffer
         inflateEnd( & mZstream );
         qDebug() << "recv Z_STREAM_END, ending compression";
         this->mNeedDecompression = false;
-        this->mMCCP_version_1 = false;
-        this->mMCCP_version_2 = false;
-
-        // Reset the option state so we can re-enable compression again in the future
-        // such as in the case of a copyover -JM
-        hisOptionState[static_cast<int>(OPT_COMPRESS)] = false;
-        hisOptionState[static_cast<int>(OPT_COMPRESS2)] = false;
```

Comment:
Are you _sure_ you should be deleting these? if I understand (and I am not sure that I do 🤕 ) the correct procedure ("Q Method" ?) for negotiating Telnet Sub-Options (i.e. [RFC1143](https://tools.ietf.org/html/rfc1143)) it means we should be tracking the option state for all options and this is part of that...

I think we are also broken on some other options as well but the whole Telnet option negotiating thingy is something I still have to get my head around.